go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -1925,6 +1945,35 @@ func (s *VtctldServer) ShardReplicationPositions(ctx context.Context, req *vtctl
 	}, nil
 }
 
+// SleepTablet is part of the vtctlservicepb.VtctldServer interface.
+func (s *VtctldServer) SleepTablet(ctx context.Context, req *vtctldatapb.SleepTabletRequest) (*vtctldatapb.SleepTabletResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "VtctldServer.SleepTablet")
+	defer span.Finish()
+
+	span.Annotate("tablet_alias", topoproto.TabletAliasString(req.TabletAlias))
+
+	dur, ok, err := protoutil.DurationFromProto(req.Duration)
+	if err != nil {
+		return nil, err
+	} else if !ok {
+		dur = *topo.RemoteOperationTimeout
```

Comment:
I'm kinda surprised that we don't return if we can't parse the duration. Does using `RemoteOperationTimeout` here mean "sleep for the maximum timeout"? Or will `tmc.Sleep` fail? 

(I did a bit of hunting in the codebase to answer this for myself but wound up in an interface declaration, and `Sleep` is... a cumbersome search term, haha.)

Reply:
the three-tuple here is `(parsedDuration, wasDurationSet, error)`, so if the duration is `nil` you get back `0, false, nil`, and we default to `*topo.RemoteOperationTimeout`. If we actually fail to parse, then the third value is non-nil and we take the earlier case (`err != nil`)

Reply:
(it's named "ok" to mirror the map two-value semantic of (value, isThereAValue))